File: FrontEndAngular/src/app/models/comment.ts
export class Comment {
  id: number;
  content: string;
  timestamp: Date;
  visiteur_id: number;
  recipe_id: number;

  constructor(
    id: number,
    content: string,
    timestamp: Date,
    visiteur_id: number,
    recipe_id: number,

  ) {
    this.id = id;
    this.content = content;
    this.timestamp = timestamp;
    this.visiteur_id = visiteur_id;
    this.recipe_id = recipe_id;

  }
}

File: conseil/src/main/java/com/example/conseil/entities/Comment.java
package com.example.conseil.entities;

import jakarta.persistence.*;
import jakarta.validation.constraints.NotBlank;
import lombok.*;

import java.sql.Date;
import java.time.LocalDate;
import java.time.LocalDateTime;

@Entity
@Table(name = "comments")
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Builder
public class Comment {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @NotBlank(message = "Le contenu ne peut pas être vide")
    @Column(nullable = false)
    private String content;

    @Column(nullable = false)
    private Date timestamp;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "visiteur_id", nullable = false)
    private Visiteur visiteur;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "recipe_id")
    private Recipe recipe;

}